### PR TITLE
FeatureView overwrites feature in source

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

FeatureView overwrites the feature in the source if they have the same name when using FlinkProcessor

## Brief change log

- Update FlinkProcessor to overwrite the feature in the source if the feature in FeatureView has the same name.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Sxnan** [comment]
@lindong28 Could you review this PR?

File: python/feathub/feature_views/transforms/tests/test_expression_transform.py [hunk 1]
```diff
@@ -268,3 +268,74 @@ def _test_case(self, upper_name_feature: Feature):
...
+        expected_result_df["cost"] = expected_result_df.apply(
+            lambda row: row["cost"] + 1, axis=1
+        )
+        expected_result_df = expected_result_df.sort_values(by=["time"]).reset_index(
+            drop=True
+        )
+
+        self.assertTrue(expected_result_df.equals(result_df))
+
+    def test_refer_to_source_feature(self):
```
Comment (lindong28) [inline python/feathub/feature_views/transforms/tests/test_expression_transform.py:303]: It seems that this test includes the FeatHub program used in `test_duplicate_feature_name_with_source`. Would it be simpler to check the result of `feature_view_1` in this test and remove the other test?

The name `test_duplicate_feature_name_with_source` is probably better.

File: python/feathub/feature_views/tests/test_feature_view.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def test_features(self):
         )
         built_feature_view = self.registry.build_features([feature_view_2])[0]
 
-        self.assertEqual(feature_1, built_feature_view.get_feature("feature_1"))
+        self.assertIsNotNone(built_feature_view.get_feature("feature_1"))
```
Comment (lindong28) [inline python/feathub/feature_views/tests/test_feature_view.py:81]: If it is not equal `feature_1`, what would be the expected value? Can we specify the expected value here?

File: python/feathub/feature_views/derived_feature_view.py [hunk 3]
```diff
@@ -47,7 +50,10 @@ def __init__(
                          feature is a string, it should be either in the format
                          {table_name}.{feature_name}, which refers to a feature in the
                          table with the given name, or in the format {feature_name},
-                         which refers to a feature in the source table. For any feature
+                         which refers to a feature in the source table. If
```
Comment (lindong28) [inline python/feathub/feature_views/derived_feature_view.py:53]: Even if we don't add this paragraph in the doc, the existing doc already says that we will compute these `features`. So the doc seems redundant here, right?

Should we update the doc for `keep_source_fields`?

File: python/feathub/feature_views/derived_feature_view.py [hunk 4]
```diff
@@ -98,7 +104,8 @@ def build(
                 table_desc = registry.get_features(name=join_table_name)
                 join_feature = table_desc.get_feature(feature_name=join_feature_name)
                 if source.name == join_table_name:
-                    feature = join_feature
+                    feature = deepcopy(join_feature)
```
Comment (lindong28) [inline python/feathub/feature_views/derived_feature_view.py:107]: Can you explain why we should copy the `input_features` from the `join_features`?

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR. Left some comments below.

**Comment by Sxnan** [comment]
@lindong28 Thanks for your review. I updated the PR. Could you please take another look?

File: python/feathub/feature_views/derived_feature_view.py (hunk 4 above)
Comment (lindong28) [reply python/feathub/feature_views/derived_feature_view.py]: Is this comment addressed?

Comment (Sxnan) [reply python/feathub/feature_views/derived_feature_view.py]: My bad. I misunderstood the comment and addressed the wrong problem. I think it is resolved now.

**Comment by Sxnan** [comment]
@lindong28 I updated the PR to address the last comment. Could you please take another look?

File: python/feathub/feature_views/derived_feature_view.py [hunk 5]
```diff
@@ -125,6 +107,44 @@ def build(
...
+    ) -> Feature:
+        parts = feature_str.split(".")
+        if len(parts) == 1:
+            source_feature = source.get_feature(parts[0])
+            feature = Feature(
+                name=source_feature.name,
+                dtype=source_feature.dtype,
+                transform=ExpressionTransform(source_feature.name),
+                keys=source_feature.keys,
+                input_features=(),
```
Comment (lindong28) [inline python/feathub/feature_views/derived_feature_view.py:122]: nits: it seems redundant to specify the default value for `input_features`.

File: python/feathub/feature_views/sliding_feature_view.py [hunk 6]
```diff
@@ -209,7 +209,14 @@ def build(
...
         for feature in self.features:
             if isinstance(feature, str):
-                feature = source.get_feature(feature_name=feature)
+                source_feature = source.get_feature(feature)
+                feature = Feature(
+                    name=source_feature.name,
+                    dtype=source_feature.dtype,
+                    transform=ExpressionTransform(source_feature.name),
+                    keys=source_feature.keys,
+                    input_features=(),
```
Comment (lindong28) [inline python/feathub/feature_views/sliding_feature_view.py:218]: nits: it seems redundant to specify the default value for `input_features`.

**Review by lindong28 — COMMENTED** [review]
Thanks for the update. Left some minor comments.

**Comment by Sxnan** [comment]
@lindong28 Thanks for your review. I updated the PR. Could you please take another look?

**Comment by lindong28** [comment]
Thanks for the update. LGTM.

**Pull request merged**